src/TestingUtils/Microsoft.AspNetCore.Testing/test/FlakyAttributeTest.cs

```diff
@@ -0,0 +1,59 @@
+using Microsoft.AspNetCore.Testing.xunit;
+using System;
+using System.Collections.Generic;
+using Xunit;
+
+namespace Microsoft.AspNetCore.Testing.Tests
+{
+    public class FlakyAttributeTest
+    {
+        [Fact]
+        [Flaky("http://example.com")]
+        public void AlwaysFlaky()
+        {
+            throw new Exception("Flakey!");
+        }
+
+        [Fact]
+        [Flaky("http://example.com", OnAzDO = false)]
+        public void FlakyInHelixOnly()
+        {
+            if (!string.IsNullOrEmpty(Environment.GetEnvironmentVariable("HELIX")))
+            {
+                throw new Exception("Flaky on Helix!");
+            }
+        }
+
+        [Fact]
+        [Flaky("http://example.com", OnAzDO = false, OnHelixQueues = HelixQueues.macOS1012Amd64 + HelixQueues.Fedora28Amd64)]
+        public void FlakyInSpecificHelixQueue()
+        {
+            var queueName = Environment.GetEnvironmentVariable("HELIX");
+            if (!string.IsNullOrEmpty(queueName))
+            {
+
+                // Normalize the queue name to have a trailing ';' (this is only for testing anyway)
+                if (!queueName.EndsWith(";"))
+                {
+                    queueName = $"{queueName};";
+                }
+
+                var failingQueues = new HashSet<string>(StringComparer.OrdinalIgnoreCase) { HelixQueues.macOS1012Amd64, HelixQueues.Fedora28Amd64 };
+                if (failingQueues.Contains(queueName))
+                {
+                    throw new Exception("Flaky on Helix!");
+                }
+            }
+        }
+
+        [Fact]
+        [Flaky("http://example.com", OnHelixQueues = HelixQueues.None)]
```

Comment:
Can we use these on a few existing tests we know are flaky? Start putting the badge of shame up.

Reply:
Not yet. We'll need build goop to actually exclude them

Reply:
So you could use it when this is in, but it won't actually do anything.

Reply:
To make it more fun, it looks like Extensions and AspNetCore will need slightly different build goop, but it looks doable.

Reply:
The good news is that helix isn't used on extensions... so you won't be getting many flaky helix tests on there hopefully :)